### DISCUSSION TITLE
关于从stak中取值出现异常的问题

### DISCUSSION THREAD

**Discussion by LunarJune** [arcsysu/SYsU-lang#30] [Q&A] [answered]
问题代码段如下：
![image](https://user-images.githubusercontent.com/61147356/163212588-25bd1c6a-848d-40b8-84bd-a4cbae2cda9a.png)
理论上，输出的两个number应该相同，然而，实际输出时第二个number的值变成了direct_declarator中的value值，但是地址依旧是assignment_expression的地址不变，输出如下：
![image](https://user-images.githubusercontent.com/61147356/163212997-188491a8-d611-4d47-a115-8da5f32e9b23.png)
不知这是什么原因造成的

**Comment by wu-kan** [comment, marked as answer]
看[这条](https://github.com/arcsysu/SYsU-lang/discussions/15)
